java.sql.SQLException: Cannot get a connection, pool error Timeout waiting for idle object

```
2018/06/25 07:58:56.690 ERROR - [com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$$anonfun$$nestedInanonfun$tryAtFixedDelay$1$1] Catching
java.sql.SQLException: Cannot get a connection, pool error Timeout waiting for idle object
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:142) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.Commons2ConnectionPool.borrow(Commons2ConnectionPool.scala:48) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.DB$.readOnly(DB.scala:174) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.executeAndLoadSubmission(SqlClient.scala:61) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.reload(SqlClient.scala:475) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.hasNext(SqlClient.scala:461) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach$(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.foreach(SqlClient.scala:452) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size(TraversableOnce.scala:107) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size$(TraversableOnce.scala:105) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.size(SqlClient.scala:452) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.runner.NewerSubmissionScrapingRunner.scrapeOnePage(NewerSubmissionScrapingRunner.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.Main$.$anonfun$main$1(Main.scala:54) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.util.Try$.apply(Try.scala:209) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$.$anonfun$tryAtFixedDelay$1(ScheduledExecutorServiceExtension.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.util.NoSuchElementException: Timeout waiting for idle object
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:134) ~[atcoder-problems-assembly-0.1.jar:0.1]
	... 23 more

```

```
2018/07/06 15:24:14.359 INFO  - [com.kenkoooo.atcoder.runner.NewerSubmissionScrapingRunner] 40 submissions overlapped in arc030
2018/07/06 15:24:14.870 INFO  - [com.kenkoooo.atcoder.scraper.SubmissionScraper] scraping https://beta.atcoder.jp/contests/rco-contest-2018-final/submissions?page=44...
2018/07/06 15:24:14.911 INFO  - [com.kenkoooo.atcoder.scraper.SubmissionScraper] scraping https://beta.atcoder.jp/contests/ijpc2012pr/submissions?page=1...
2018/07/06 15:24:16.196 INFO  - [com.kenkoooo.atcoder.db.SqlClient] loading rated_point_sum
2018/07/06 15:24:16.216 INFO  - [com.kenkoooo.atcoder.runner.NewerSubmissionScrapingRunner] 20 submissions overlapped in ijpc2012pr
2018/07/06 15:24:16.726 INFO  - [com.kenkoooo.atcoder.scraper.SubmissionScraper] scraping https://beta.atcoder.jp/contests/jag2013summer-day3/submissions?page=1...
2018/07/06 15:24:19.406 INFO  - [com.kenkoooo.atcoder.db.SqlClient] loading language_count
2018/07/06 15:24:19.408 INFO  - [com.kenkoooo.atcoder.scraper.SubmissionScraper] scraping https://beta.atcoder.jp/contests/ijpc2012pr/submissions?page=2...
2018/07/06 15:24:24.453 ERROR - [com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$$anonfun$$nestedInanonfun$tryAtFixedDelay$1$1] Catching
java.sql.SQLException: Cannot get a connection, pool error Timeout waiting for idle object
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:142) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.Commons2ConnectionPool.borrow(Commons2ConnectionPool.scala:48) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.DB$.readOnly(DB.scala:174) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.executeAndLoadSubmission(SqlClient.scala:61) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.reload(SqlClient.scala:478) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.hasNext(SqlClient.scala:464) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach$(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.foreach(SqlClient.scala:455) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size(TraversableOnce.scala:107) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size$(TraversableOnce.scala:105) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.size(SqlClient.scala:455) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.runner.NewerSubmissionScrapingRunner.scrapeOnePage(NewerSubmissionScrapingRunner.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.Main$.$anonfun$main$1(Main.scala:54) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.util.Try$.apply(Try.scala:209) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$.$anonfun$tryAtFixedDelay$1(ScheduledExecutorServiceExtension.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.util.NoSuchElementException: Timeout waiting for idle object
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:134) ~[atcoder-problems-assembly-0.1.jar:0.1]
	... 23 more
2018/07/06 15:24:24.863 INFO  - [com.kenkoooo.atcoder.db.SqlClient] loading predicted_rating
2018/07/06 15:24:24.864 ERROR - [com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$$anonfun$$nestedInanonfun$tryAtFixedDelay$1$1] Catching
java.sql.SQLException: Cannot get a connection, general error
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:146) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.Commons2ConnectionPool.borrow(Commons2ConnectionPool.scala:48) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.DB$.readOnly(DB.scala:174) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.loadRecords(SqlClient.scala:353) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.reloadRecords(SqlClient.scala:339) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.Main$.$anonfun$main$2(Main.scala:58) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.util.Try$.apply(Try.scala:209) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$.$anonfun$tryAtFixedDelay$1(ScheduledExecutorServiceExtension.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireInterruptibly(AbstractQueuedSynchronizer.java:1220) ~[?:1.8.0_171]
	at java.util.concurrent.locks.ReentrantLock.lockInterruptibly(ReentrantLock.java:335) ~[?:1.8.0_171]
	at org.apache.commons.pool2.impl.LinkedBlockingDeque.pollFirst(LinkedBlockingDeque.java:624) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:444) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:134) ~[atcoder-problems-assembly-0.1.jar:0.1]
	... 15 more

```

```
java.sql.SQLException: Cannot get a connection, pool error Timeout waiting for idle object
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:142) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.Commons2ConnectionPool.borrow(Commons2ConnectionPool.scala:48) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.DB$.readOnly(DB.scala:174) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.loadRecords(SqlClient.scala:353) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.reloadRecords(SqlClient.scala:329) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.Main$.$anonfun$main$2(Main.scala:58) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.util.Try$.apply(Try.scala:209) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$.$anonfun$tryAtFixedDelay$1(ScheduledExecutorServiceExtension.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.util.NoSuchElementException: Timeout waiting for idle object
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:449) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:134) ~[atcoder-problems-assembly-0.1.jar:0.1]
	... 15 more
2018/07/10 01:53:02.774 ERROR - [com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$$anonfun$$nestedInanonfun$tryAtFixedDelay$1$1] Catching
java.sql.SQLException: Cannot get a connection, general error
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:146) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.Commons2ConnectionPool.borrow(Commons2ConnectionPool.scala:48) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scalikejdbc.DB$.readOnly(DB.scala:174) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SqlClient.executeAndLoadSubmission(SqlClient.scala:61) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.reload(SqlClient.scala:478) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.hasNext(SqlClient.scala:464) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.Iterator.foreach$(Iterator.scala:929) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.foreach(SqlClient.scala:455) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size(TraversableOnce.scala:107) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.collection.TraversableOnce.size$(TraversableOnce.scala:105) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.db.SubmissionIterator.size(SqlClient.scala:455) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.runner.NewerSubmissionScrapingRunner.scrapeOnePage(NewerSubmissionScrapingRunner.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.Main$.$anonfun$main$1(Main.scala:54) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at scala.util.Try$.apply(Try.scala:209) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at com.kenkoooo.atcoder.common.ScheduledExecutorServiceExtension$TryAndDieScheduledExecutorService$.$anonfun$tryAtFixedDelay$1(ScheduledExecutorServiceExtension.scala:30) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.reportInterruptAfterWait(AbstractQueuedSynchronizer.java:2014) ~[?:1.8.0_171]
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2088) ~[?:1.8.0_171]
	at org.apache.commons.pool2.impl.LinkedBlockingDeque.pollFirst(LinkedBlockingDeque.java:631) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:444) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:363) ~[atcoder-problems-assembly-0.1.jar:0.1]
	at org.apache.commons.dbcp2.PoolingDataSource.getConnection(PoolingDataSource.java:134) ~[atcoder-problems-assembly-0.1.jar:0.1]
	... 23 more
```

--- a/atcoder-problems-backend/src/main/scala/com/kenkoooo/atcoder/common/ScheduledExecutorServiceExtension.scala
+++ b/atcoder-problems-backend/src/main/scala/com/kenkoooo/atcoder/common/ScheduledExecutorServiceExtension.scala
@@ -31,7 +31,6 @@ object ScheduledExecutorServiceExtension extends Logging {
           }.recover {
             case e: Throwable =>
               logger.catching(e)
-              self.shutdownNow()
         },
         initialDelay,
         period,

--- a/atcoder-problems-backend/src/test/scala/com/kenkoooo/atcoder/common/ScheduledExecutorServiceExtensionTest.scala
+++ b/atcoder-problems-backend/src/test/scala/com/kenkoooo/atcoder/common/ScheduledExecutorServiceExtensionTest.scala
@@ -17,7 +17,7 @@ class ScheduledExecutorServiceExtensionTest extends FunSuite with Matchers {
     }
 
     Thread.sleep(3000)
-    service.isShutdown shouldBe true
-    x shouldBe 2
+    service.isShutdown shouldBe false
+    service.shutdown()
   }
 }

